src/main/java/com/google/flourbot/execution/MacroExecutionModuleImplementation.java

```diff
@@ -66,15 +67,25 @@ public String execute(String userEmail, String message) throws IOException, Gene
         SheetAppendAction a = (SheetAppendAction) optionalMacro.get().getAction();
         String documentId = a.getSheetId();
         DriveClient cdc = new DriveClient();
-        CloudSheet cs = cdc.getCloudSheet(documentId);
-        cs.appendRow(values);
+        CloudSheet cloudSheet = cdc.getCloudSheet(documentId);
+        cloudSheet.appendRow(values);
+        break;
+      case SHEET_READ_ROW:
+        // TODO: after merge with master, move CloudSheet init to before the switch statement
+        CloudSheet cloudSheet = cloudDocClient.getCloudSheet(documentId);
+        List<String> sheetData = cloudSheet.readRow(2); //TODO: Replace hardcoded
+        break;
+      case SHEET_READ_COLUMN:
+        CloudSheet cloudSheet = cloudDocClient.getCloudSheet(documentId);
+        List<String> sheetData = cloudSheet.readCol("B");//TODO: Replace hardcoded
         break;
       default:
         throw new IllegalStateException(
             "Action type named: " + actionType.toString() + "is not implemented yet!");
     }
 
     // TODO: Return a response object
+    // If read row or read col - give the row or col
```

Comment:
I assume we may want to implement some post processing as well, e.g. formatting with commas.

Reply:
Yes!